[CI] MlDistributedFailureIT testClusterWithTwoMlNodes_RunsDatafeed_GivenOriginalNodeGoesDown failing

Note this is a test for the arm compatibility suite. So maybe it is specific to arm?

**Build scan:**
https://gradle-enterprise.elastic.co/s/xomhct56olfds/tests/:x-pack:plugin:ml:internalClusterTest/org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT/testClusterWithTwoMlNodes_RunsDatafeed_GivenOriginalNodeGoesDown

**Reproduction line:**
`./gradlew ':x-pack:plugin:ml:internalClusterTest' --tests "org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT.testClusterWithTwoMlNodes_RunsDatafeed_GivenOriginalNodeGoesDown" -Dtests.seed=F4DE3543E5B4A1A4 -Dtests.locale=zh-Hant-TW -Dtests.timezone=Japan -Druntime.java=16`

**Applicable branches:**
master

**Reproduces locally?:**
No

**Failure history:**
https://gradle-enterprise.elastic.co/scans/tests?tests.container=org.elasticsearch.xpack.ml.integration.MlDistributedFailureIT&tests.test=testClusterWithTwoMlNodes_RunsDatafeed_GivenOriginalNodeGoesDown

**Failure excerpt:**
```
org.elasticsearch.action.FailedNodeException: Failed node [53VPJVs8S3aJgnSPycZZeg]

  at __randomizedtesting.SeedInfo.seed([F4DE3543E5B4A1A4:539301513C11154F]:0)
  at org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction.onFailure(TransportTasksAction.java:283)
  at org.elasticsearch.action.support.tasks.TransportTasksAction$AsyncAction$1.handleException(TransportTasksAction.java:262)
  at org.elasticsearch.transport.TransportService$5.handleException(TransportService.java:629)
  at org.elasticsearch.transport.TransportService$ContextRestoreResponseHandler.handleException(TransportService.java:1173)
  at org.elasticsearch.transport.InboundHandler.lambda$handleException$3(InboundHandler.java:313)
  at org.elasticsearch.common.util.concurrent.EsExecutors$DirectExecutorService.execute(EsExecutors.java:178)
  at org.elasticsearch.transport.InboundHandler.handleException(InboundHandler.java:311)
  at org.elasticsearch.transport.InboundHandler.handlerResponseError(InboundHandler.java:303)
  at org.elasticsearch.transport.InboundHandler.messageReceived(InboundHandler.java:124)
  at org.elasticsearch.transport.InboundHandler.inboundMessage(InboundHandler.java:82)
  at org.elasticsearch.transport.TcpTransport.inboundMessage(TcpTransport.java:695)
  at org.elasticsearch.transport.InboundPipeline.forwardFragments(InboundPipeline.java:129)
  at org.elasticsearch.transport.InboundPipeline.doHandleBytes(InboundPipeline.java:104)
  at org.elasticsearch.transport.InboundPipeline.handleBytes(InboundPipeline.java:69)
  at org.elasticsearch.transport.nio.MockNioTransport$MockTcpReadWriteHandler.consumeReads(MockNioTransport.java:312)
  at org.elasticsearch.nio.SocketChannelContext.handleReadBytes(SocketChannelContext.java:217)
  at org.elasticsearch.nio.BytesChannelContext.read(BytesChannelContext.java:29)
  at org.elasticsearch.nio.EventHandler.handleRead(EventHandler.java:128)
  at org.elasticsearch.transport.nio.TestEventHandler.handleRead(TestEventHandler.java:140)
  at org.elasticsearch.nio.NioSelector.handleRead(NioSelector.java:409)
  at org.elasticsearch.nio.NioSelector.processKey(NioSelector.java:235)
  at org.elasticsearch.nio.NioSelector.singleLoop(NioSelector.java:163)
  at org.elasticsearch.nio.NioSelector.runLoop(NioSelector.java:120)
  at java.lang.Thread.run(Thread.java:831)

  Caused by: org.elasticsearch.transport.RemoteTransportException: [node_t3][127.0.0.1:42083][cluster:internal/xpack/ml/datafeed/running_state[n]]


    Caused by: java.lang.NullPointerException: Cannot invoke "org.elasticsearch.xpack.core.ml.action.GetDatafeedRunningStateAction$Response$RunningState.writeTo(org.elasticsearch.common.io.stream.StreamOutput)" because "w" is null

      at org.elasticsearch.xpack.core.ml.action.GetDatafeedRunningStateAction$Response.lambda$writeTo$2(GetDatafeedRunningStateAction.java:153)
      at org.elasticsearch.common.io.stream.StreamOutput.writeMap(StreamOutput.java:586)
      at org.elasticsearch.xpack.core.ml.action.GetDatafeedRunningStateAction$Response.writeTo(GetDatafeedRunningStateAction.java:153)
      at org.elasticsearch.action.support.tasks.TransportTasksAction$NodeTasksResponse.writeTo(TransportTasksAction.java:389)
      at org.elasticsearch.transport.OutboundMessage.serialize(OutboundMessage.java:65)
      at org.elasticsearch.transport.OutboundHandler.sendMessage(OutboundHandler.java:135)
      at org.elasticsearch.transport.OutboundHandler.sendResponse(OutboundHandler.java:113)
      at org.elasticsearch.transport.TcpTransportChannel.sendResponse(TcpTransportChannel.java:49)
      at org.elasticsearch.transport.TaskTransportChannel.sendResponse(TaskTransportChannel.java:41)
      at org.elasticsearch.action.ActionListener$1.onResponse(ActionListener.java:134)
      at org.elasticsearch.action.support.tasks.TransportTasksAction$1.respondIfFinished(TransportTasksAction.java:127)
      at org.elasticsearch.action.support.tasks.TransportTasksAction$1.onResponse(TransportTasksAction.java:102)
      at org.elasticsearch.action.support.tasks.TransportTasksAction$1.onResponse(TransportTasksAction.java:98)
      at org.elasticsearch.xpack.ml.action.TransportGetDatafeedRunningStateAction.taskOperation(TransportGetDatafeedRunningStateAction.java:82)
      at org.elasticsearch.xpack.ml.action.TransportGetDatafeedRunningStateAction.taskOperation(TransportGetDatafeedRunningStateAction.java:36)
      at org.elasticsearch.action.support.tasks.TransportTasksAction.nodeOperation(TransportTasksAction.java:131)
      at org.elasticsearch.action.support.tasks.TransportTasksAction$NodeTransportHandler.messageReceived(TransportTasksAction.java:307)
      at org.elasticsearch.action.support.tasks.TransportTasksAction$NodeTransportHandler.messageReceived(TransportTasksAction.java:303)
      at org.elasticsearch.transport.RequestHandlerRegistry.processMessageReceived(RequestHandlerRegistry.java:61)
      at org.elasticsearch.transport.InboundHandler$1.doRun(InboundHandler.java:212)
      at org.elasticsearch.common.util.concurrent.ThreadContext$ContextPreservingAbstractRunnable.doRun(ThreadContext.java:737)
      at org.elasticsearch.common.util.concurrent.AbstractRunnable.run(AbstractRunnable.java:26)
      at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130)
      at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630)
      at java.lang.Thread.run(Thread.java:831)

```

--- a/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetDatafeedRunningStateAction.java
+++ b/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportGetDatafeedRunningStateAction.java
@@ -79,7 +79,7 @@ public class TransportGetDatafeedRunningStateAction extends TransportTasksAction
         TransportStartDatafeedAction.DatafeedTask datafeedTask,
         ActionListener<Response> listener
     ) {
-        listener.onResponse(Response.fromTaskAndState(datafeedTask.getDatafeedId(), datafeedTask.getRunningState().orElse(null)));
+        listener.onResponse(Response.fromTaskAndState(datafeedTask.getDatafeedId(), datafeedTask.getRunningState()));
     }
 
     @Override

--- a/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartDatafeedAction.java
+++ b/x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/action/TransportStartDatafeedAction.java
@@ -596,16 +596,19 @@ public class TransportStartDatafeedAction extends TransportMasterNodeAction<Star
             }
         }
 
-        public Optional<GetDatafeedRunningStateAction.Response.RunningState> getRunningState() {
+        public GetDatafeedRunningStateAction.Response.RunningState getRunningState() {
             synchronized (this) {
                 if (datafeedRunner == null) {
-                    return Optional.empty();
+                    // In this case we don't know for sure if lookback has completed.  It may be that the
+                    // datafeed has just moved nodes, but with so little delay that there's no lookback to
+                    // do on the new node.  However, there _might_ be some catching up required, so it's
+                    // reasonable to say real-time running hasn't started yet.  The state will quickly
+                    // change once the datafeed runner gets going and determines where the datafeed is up
+                    // to.
+                    return new GetDatafeedRunningStateAction.Response.RunningState(endTime == null,false);
                 }
             }
-            return Optional.of(new GetDatafeedRunningStateAction.Response.RunningState(
-                this.endTime == null,
-                datafeedRunner.finishedLookBack(this)
-            ));
+            return new GetDatafeedRunningStateAction.Response.RunningState(endTime == null, datafeedRunner.finishedLookBack(this));
         }
     }
 

